File: tlb/08/solutions/largest_numbered_line.v2.pl
#!/usr/bin/perl -w
push @numbers, /(\-?\d+(?:\.\d+)?)/g foreach @lines = <STDIN>;
exit if !@numbers;
$largest = (sort {$b <=> $a} @numbers)[0];
(sort {$b <=> $a} /(\-?\d+(?:\.\d+)?)/g)[0] == $largest && print foreach @lines;

File: tlb/08/solutions/largest_numbered_line.v2.py
#!/usr/bin/python3
import re, sys
n = [(max([float(x) for x in re.findall(r'(\-?\d+(?:\.\d+)?)', line) if x]),line) for line in sys.stdin]
print ''.join([line for (x,line) in n if x == sorted(n)[0][0]]),
